go/adbc/adbc.go

```diff
@@ -58,13 +58,15 @@ type Error struct {
 	// SqlState is a SQLSTATE error code, if provided, as defined
 	// by the SQL:2003 standard. If not set, it will be "\0\0\0\0\0"
 	SqlState [5]byte
-	// Details is an array of additional driver-specific binary error details.
+	// Details is an array of additional driver-specific error details.
 	//
 	// This allows drivers to return custom, structured error information (for
 	// example, JSON or Protocol Buffers) that can be optionally parsed by
 	// clients, beyond the standard Error fields, without having to encode it in
-	// the error message.  The encoding of the data is driver-defined.
-	Details [][]byte
+	// the error message.  The encoding of the data is driver-defined.  It is
+	// suggested to use proto.Message for Protocol Buffers and error for wrapped
+	// errors.
+	Details []interface{}
```

Comment:
why the switch to use `interface{}` instead of `[]byte`? it's a bit more difficult to convert this to something usable for a C error struct for an `interface{}`

Reply:
So would `[](pair of key, []byte)` be preferable?

Reply:
you can use the `proto.Message` interface potentially and then just serialize them later when necessary?